noderesourcetopology: add more tests and internal cleanups (#356)
kubernetes-sigs/scheduler-plugins: pkg/noderesourcetopology/README.md

        type: "LeastAllocated"
```

#### Cluster

> The Topology-aware scheduler performs its decision over a number of node-specific hardware details or configuration setting which have node granularity (not at cluster granularity).

Reviewer: ```suggestion
The Topology-aware scheduler performs its decision over a number of node-specific hardware details or configuration settings which have node granularity (not at cluster granularity).
```

Author: done

> A fundamental prerequisite of the scheduler to work correctly, is these settings are consistent across a set of nodes, or all over the cluster.

Reviewer: Not an English expert in anyway, but I think with a minor change this would read better:
```suggestion
Consistent settings across a set of nodes or all over the cluster is a fundamental prerequisite for the scheduler to work correctly.
```

Author: yes, thanks!

> In other words, the prerequisite is a set of nodes all share the same NUMA topology and kubelet configuration, at least for settings like topology and resource (cpu, memory, device) managers.

Reviewer: ```suggestion
In other words, it is a prerequisite that a set of nodes share the same NUMA topology and kubelet configuration, at least for settings like topology and resource (cpu, memory, device) managers.
```

Author: thanks, fixed

However, the scheduler has no means to enforce or even validate this prerequisite;
for example the [NodeResourceTopology](https://github.com/k8stopologyawareschedwg/noderesourcetopology-api) CRD does not expose all the relevant fields, nor it should (it would be out of scope).

> Hence, proper cluster configuration is demanded on the cluster admins, or to other software components, like controllers or operators, outside of the scope here.

Reviewer: ```suggestion
Hence, proper cluster configuration is expected from the cluster admins, or other software components, like controllers or operators, outside of the scope here.
```

Author: thanks, fixed!
